tests/conftest.py

```diff
@@ -20,3 +26,131 @@ def _random_wrapper_applier(arrays, wrapper):
         return [wrapper(arr) if rng.random() > 0.5 else arr for arr in arrays]
 
     return _random_wrapper_applier
+
+
+# Regression fixtures, helpers, and hooks
+# If you want to change the base directory, make sure to
+# update pyproject.toml, and pre-commit config to make
+# sure the references are not linted/formatted.
+base_regr_dir = Path("tests/references")
+
+
+@pytest.fixture
+def lazy_datadir(request) -> Path:
+    """
+    Creates a unique per-test directory for reference files.
+    E.g., tests/references/<module_name>/<test_name>/
+    """
+    # test_module will be something like "tests.test_logic_compiler"
+    # We don't want the "tests." part
+    module_parts = request.module.__name__.split(".")
+    if module_parts[0] == "tests" and len(module_parts) > 1:
+        test_module = "/".join(module_parts[1:])
```

Comment:
This should be done with pathlib.path. Could you also add pathlib.path to the style guide (saying we use that library instead of os.path)?